### DISCUSSION TITLE
Lazygit is laggy

### DISCUSSION THREAD

**Discussion by halfpastfive23** [jesseduffield/lazygit#3965] [Q&A]
I couldn't use the lazygit, everytime I hit a key as it lags out and I have to wait for a second or two just get the output

**Comment by mark2185** [comment]
Could you please fill out the issue template? And try to add relevant details e.g. is it in a big repo (if public, please share it), is it on windows etc.

**Reply by halfpastfive23** [reply]
Hey, my bad for not giving the context well.I'm using MacOS. It is a not a big repo as I just created them just to test it. I just fixed it but I'm still open for any solution on this as well.

**Comment by stefanhaller** [comment]
I can only guess if your problem is the same as what I saw recently. In my main work repo, which is moderately large but definitely not huge (>50'000 files, >300'000 commits), lazygit became slower and slower over the past couple weeks, until things like switching between Files and Commits panel took more than a second.

I solved it by running the following commands:
```
git maintenance run --schedule=hourly
git maintenance run --schedule=daily
git maintenance run --schedule=weekly
git maintenance run --task=gc
```
Now it's nice and fast again. I don't know which of these commands helped the most, just try them all.

Also, I'm confused as to why this was necessary, as I did enable maintenance (using `git maintenance start`) for this repo months ago, so it is supposed to run these (except the last one, I think) automatically in the background. I find it difficult to troubleshoot this, as the scheduled tasks don't seem to write to any log files, so it's hard to verify if they ran or what they did.

If anybody knows more about this, I'm all ears. For now, I'm just tucking those commands in a script which I run once week or so.

**Reply by halfpastfive23** [reply]
Hey, it indeed worked. I just paste all of it one-by-one.Appreciated and I hope that it will get fixed for long run.
